app/assets/src/components/common/ProjectCreationForm.jsx

```diff
@@ -14,42 +14,61 @@ import { logAnalyticsEvent } from "~/api/analytics";
 
 import cs from "./project_creation_form.scss";
 
+const ACCESS_LEVEL = Object.freeze({
+  publicAccess: 1,
+  privateAccess: 0,
+  noSelection: -1,
+});
+
 class ProjectCreationForm extends React.Component {
   state = {
     name: "",
-    publicAccess: -1, // No selection by default
+    accessLevel: ACCESS_LEVEL.noSelection, // No selection by default
     error: "",
     description: "",
     showInfo: false,
+    disableCreateButton: true,
   };
 
   handleNameChange = name => {
+    const disableCreateButton = name === "";
     this.setState({
       name,
+      disableCreateButton,
+    });
+  };
+
+  handleAccessLevelChange = accessLevel => {
+    const disableCreateButton = accessLevel === -1;
```

Comment:
Doesn't `disableCreateButton` always depend on the all conditions: accessLevel and disableCreateButton (not name length)? You cannot check them just one at a time, right?

Reply:
Yes, thank you for catching.